internal/adapters/dataproviders/database/mongo/pegin.go

```diff
@@ -68,6 +68,39 @@ func (repo *peginMongoRepository) GetQuote(ctx context.Context, hash string) (*q
 	return &result.PeginQuote, nil
 }
 
+func (repo *peginMongoRepository) GetQuotes(ctx context.Context, hashes []string) ([]quote.PeginQuote, error) {
+	var result StoredPeginQuote
+	dbCtx, cancel := context.WithTimeout(ctx, repo.conn.timeout)
+	defer cancel()
+
+	for _, hash := range hashes {
+		if err := quote.ValidateQuoteHash(hash); err != nil {
+			return nil, err
+		}
+	}
+
+	collection := repo.conn.Collection(PeginQuoteCollection)
+	filter := bson.M{"hash": bson.M{"$in": hashes}}
+
+	quotesReturn := make([]quote.PeginQuote, 0)
+
+	cursor, err := collection.Find(dbCtx, filter)
+	if errors.Is(err, mongo.ErrNoDocuments) {
+		return nil, nil
```

Comment:
I'd return the empty array in here, if you don't include an error, the consumer will asume the result has something in it